hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JetBootstrap.java

```diff
@@ -157,15 +143,44 @@ private static void error(String msg) {
      * Returns the bootstrapped {@code JetInstance}. The instance will be
      * automatically shut down once the {@code main()} method of the JAR returns.
      */
-    public static JetInstance getInstance() {
+    @Nonnull
+    public static synchronized JetInstance getInstance() {
         if (supplier == null) {
-            throw new JetException(
-                    "JetBootstrap.getInstance() should be used in conjunction with the jet submit command"
-            );
+            supplier = memoizeConcurrent(JetBootstrap::createStandaloneInstance);
         }
         return supplier.get();
     }
 
+    private static JetInstance createStandaloneInstance() {
+        configureLogging();
+        LOGGER.info("Bootstrapped instance requested but application wasn't called from jet submit script. " +
+                "Creating a standalone Jet instance instead.");
+        JetConfig config = locateAndGetJetConfig();
+        Config hzconfig = config.getHazelcastConfig();
+
+        // turn off all discovery to make sure node doesn't join any existing cluster
+        hzconfig.setProperty("hazelcast.wait.seconds.before.join", "0");
+        JoinConfig join = hzconfig.getNetworkConfig().getJoin();
+        join.getMulticastConfig().setEnabled(false);
+        join.getTcpIpConfig().setEnabled(false);
+        join.getAwsConfig().setEnabled(false);
+        join.getGcpConfig().setEnabled(false);
+        join.getAzureConfig().setEnabled(false);
+        join.getKubernetesConfig().setEnabled(false);
+        join.getEurekaConfig().setEnabled(false);
```

Comment:
we may need to disable DiscoveryConfig too by setting an empty config
`join.setDiscoveryConfig(new DiscoveryConfig());`
And disable advanced network config
`hzconfig.getAdvancedNetworkConfig().setEnabled(false);`